crates/python/qcs_sdk/__init__.pyi

```diff
@@ -1,29 +1,357 @@
+
+        :raises ExecutionError: If the job fails to execute.
+        """
+        ...
+    async def execute_on_qpu_async(
+        self, quantum_processor_id: str, endpoint_id: Optional[str] = None
+    ) -> ExecutionData:
+        """
+        Compile the program and execute it on a QPU, waiting for results.
+        (async analog of ``Executable.execute_on_qpu``)
+
+        :param endpoint_id: execute the compiled program against an explicitly provided endpoint. If `None`,
+        the default endpoint for the given quantum_processor_id is used.
+
+        :raises ExecutionError: If the job fails to execute.
+        """
+        ...
+    def submit_to_qpu(
+        self, quantum_processor_id: str, endpoint_id: Optional[str] = None
+    ) -> JobHandle:
+        """
+        Compile the program and execute it on a QPU, without waiting for results.
+
+        :param endpoint_id: execute the compiled program against an explicitly provided endpoint. If `None`,
+        the default endpoint for the given quantum_processor_id is used.
+
+        :raises ExecutionError: If the job fails to execute.
+        """
+        ...
+    async def submit_to_qpu_async(
+        self, quantum_processor_id: str, endpoint_id: Optional[str] = None
+    ) -> JobHandle:
+        """
+        Compile the program and execute it on a QPU, without waiting for results.
+        (async analog of ``Executable.execute_on_qpu``)
+
+        :param endpoint_id: execute the compiled program against an explicitly provided endpoint. If `None`,
+        the default endpoint for the given quantum_processor_id is used.
+
+        :raises ExecutionError: If the job fails to execute.
+        """
+        ...
+    def retrieve_results(self, job_handle: JobHandle) -> ExecutionData:
+        """
+        Wait for the results of a job to complete.
+
+        :raises ExecutionError: If the job fails to execute.
+        """
+        ...
+    async def retrieve_results_async(self, job_handle: JobHandle) -> ExecutionData:
+        """
+        Wait for the results of a job to complete.
+        (async analog of ``Executable.retrieve_results``)
+
+        :raises ExecutionError: If the job fails to execute.
+        """
+        ...
+
+@final
+class JobHandle:
+    """
+    The result of submitting a job to a QPU.
+
+    Used to retrieve the results of a job.
+    """
+
+    @property
+    def job_id(self) -> str:
+        """
+        Unique ID associated with a single job execution.
+        """
+        ...
+    @property
+    def readout_map(self) -> Dict[str, str]:
+        """
+        The readout map from source readout memory locations to the filter pipeline node which publishes the data.
+        """
+        ...
+
+@final
+class ExeParameter:
+    """
+    Program execution parameters.
+
+    Note: The validity of parameters is not checked until execution.
+    """
+
+    def __new__(
+        cls: type["ExeParameter"],
```

Comment:
I think the type annotation here can be omitted, depending on the language server - doesn't hurt I guess though